### ISSUE TITLE
[AVX10.2]Optimize codegen for x86 using the new saturating conversion instructions when possible.

### ISSUE THREAD

**Issue by DeepakRajendrakumaran** [dotnet/runtime#109080]
`AVX10.2` introduces a set of saturating conversion instructions. These will allow us to further optimize the various floating point-> integer conversions. Specifically cases where we are currently handling edge cases with additional instructions. With these new instructions, we can accomplish this with just 1 instruction

Eg. `double` -> `ulong`

This currently generates the following 2 instructions

```
vfixupimmsd 
vcvttsd2usi
```

The new Avx10.2 instructions are the following
![Image](https://github.com/user-attachments/assets/1fc78207-c293-4ef8-b536-c7aeff56dcba)

**Comment by JulieLeeMSFT** [comment]
@khushal1996, can we close this issue becuase @111775 is merged now?

**Comment by khushal1996** [comment]
@JulieLeeMSFT  yes. we can close this now and the work related to this issue has been merged.
